frontend/src/metabase/visualizations/components/TableInteractive/TableInteractive.tsx

```diff
@@ -445,7 +445,9 @@ export const TableInteractiveInner = forwardRef(function TableInteractiveInner(
             value,
             rowIndex,
             col.name,
-          ) ?? tableTheme?.cell?.backgroundColor,
+          ) ??
+          (tableTheme?.cell?.backgroundColor &&
+            theme.fn.themeColor(tableTheme?.cell?.backgroundColor)),
```

Comment:
- We cannot use CSS variables like `var(--mb-background-color)` here otherwise the cell background throws an error.
- This `theme.fn.themeColor` allows us to pass `"background"` to it when on the SDK, and users can still pass hex and rgb codes like `#2d2d30` without issues.
- This only applies `background` on the SDK, so it remains compatible with static embedding by falling back to `undefined` so it just becomes transparent and use the night theme background color